Select the right default version for a dialect  when  using Hibernate Reactive

DavideD opened the pull request:
Fix #34032 

It also adds a new `integration-tests/hibernate-reactive-mssql` that's similar to `integration-tests/jpa-mssql`.

DavideD committed:
[#34032] Select correct defaul dialect and storage engine for HR

Similar to what happens in `HibernateORMProcessor#collectDialectConfig`.

Hibernate Reactiv will select the dialect default version using the
values in the class `io.quarkus.hibernate.orm.runtime.config.DialectVersions`.

yrodiere commented on integration-tests/hibernate-reactive-mssql/src/main/java/io/quarkus/it/jpa/mssql/HibernateReactiveFunctionalityTestEndpoint.java:
```diff
@@ -0,0 +1,179 @@
+        System.out.print(sb);
+    }
+
+    private static void persistNewPerson(EntityManager entityManager, String name) {
+        Person person = new Person();
+        person.setName(name);
+        person.setAddress(new SequencedAddress("Street " + randomName()));
+        entityManager.persist(person);
+    }
```
These tests are using JPA instead of the Hibernate Reactive API... ?

yrodiere commented on integration-tests/hibernate-reactive-mssql/src/test/java/io/quarkus/it/jpa/mssql/DeafultDialectVersionTest.java:
```diff
@@ -0,0 +1,27 @@
+ * Test the dialect used by default in Quarkus.
+ */
+@QuarkusTest
+public class DeafultDialectVersionTest {
```
Typo in the class name.

yrodiere commented on integration-tests/hibernate-reactive-mssql/src/test/java/io/quarkus/it/jpa/mssql/DeafultDialectVersionTest.java:
FYI, for Hibernate ORM we duplicated this test for every supported DB. I see we already have integration tests for Hibernate Reactive with postgres, db2 and mysql; you may want to copy/paste the test and corresponding endpoint there.

yrodiere commented on integration-tests/hibernate-reactive-mssql/README.md:
```diff
@@ -0,0 +1,47 @@
+# JPA example with Microsoft SQL Server
```
IMO we should either remove the README or update it to talk about HR instead of JPA.

gsmet replied:
I would drop it.

yrodiere reviewed (dismissed):
Thanks! The implementation looks good, but the test is using JPA, so that looks wrong to me.

yrodiere commented on integration-tests/hibernate-reactive-mssql/src/main/java/io/quarkus/it/hibernate/reactive/mssql/HibernateReactiveMySQLAlternativeTestEndpoint.java:
```diff
@@ -0,0 +1,101 @@
+import jakarta.ws.rs.Path;
+
+@Path("/alternative-tests")
+public class HibernateReactiveMySQLAlternativeTestEndpoint {
```
This is the MS SQL module, though... not MySQL.

yrodiere commented on integration-tests/hibernate-reactive-mssql/src/main/java/io/quarkus/it/hibernate/reactive/mssql/HibernateReactiveMySQLTestEndpoint.java:
```diff
@@ -0,0 +1,100 @@
+import jakarta.ws.rs.Path;
+
+@Path("/tests")
+public class HibernateReactiveMySQLTestEndpoint {
```
This is the MS SQL module, though... not MySQL.

yrodiere commented on integration-tests/hibernate-reactive-mssql/src/main/java/io/quarkus/it/hibernate/reactive/mssql/HibernateReactiveMySQLAlternativeTestEndpoint.java:
As far as I can see, the only difference between `HibernateReactiveMySQLAlternativeTestEndpoint` `HibernateReactiveMySQLTestEndpoint` is the name of the class and some formatting... Are you sure this is what you wanted?

DavideD replied:
I've just copied all the classes from the MySQL module (And I forgot to rename a couple of classes). 
I guess the integration tests need some clean up everywhere.

DavideD replied:
Ah, these alternative endpoint were introduced when it was possible to use the `CompletionStage` API. I will remove them

yrodiere reviewed (commented):
Thanks. Tests look fine, but I think there are some oversights. I'll let you decide what you want to do...

DavideD committed:
[#34032] Remove alternative end-points from integration-tests

Their purpose was to test the CompletionStage API for Hibernate Reactive
in Quarkus. But, we decided some time ago
to only allow the Mutiny API.

Insted of removing the classes, we changed them to use Mutiny, ending
with the same tests we have in the non alternative end-point.

DavideD commented:
I've renamed the classes and remove the tests with the alternative end-point

yrodiere reviewed (approved):
LGTM, thanks!
